Signal/src/ViewControllers/DebugUI/DebugUIMessages.m

```diff
@@ -1108,6 +1123,149 @@ + (void)sendFakeMessages:(NSUInteger)counter
+
+                [self sendFakeMediaMessages:counter
+                                     thread:thread
+                              gifDataSource:gifDataSource
+                             jpegDataSource:jpegDataSource
+                              mp4DataSource:mp4DataSource];
+            }
+                                     failure:failureBlock];
+        }
+                                  failure:failureBlock];
+    }
+                             failure:failureBlock];
+}
+
++ (void)sendFakeMediaMessages:(NSUInteger)counter
+                       thread:(TSThread *)thread
+                gifDataSource:(DataSource *)gifDataSource
+               jpegDataSource:(DataSource *)jpegDataSource
+                mp4DataSource:(DataSource *)mp4DataSource
+{
+    const NSUInteger kMaxBatchSize = 2500;
+    if (counter < kMaxBatchSize) {
+        [OWSPrimaryStorage.dbReadWriteConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+            [self sendFakeMediaMessages:counter
+                                 thread:thread
+                          gifDataSource:gifDataSource
+                         jpegDataSource:jpegDataSource
+                          mp4DataSource:mp4DataSource
+                            transaction:transaction];
+        }];
+    } else {
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+            NSUInteger remainder = counter;
+            while (remainder > 0) {
+                NSUInteger batchSize = MIN(kMaxBatchSize, remainder);
+                [OWSPrimaryStorage.dbReadWriteConnection
+                    readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+                        [self sendFakeMediaMessages:batchSize
+                                             thread:thread
+                                      gifDataSource:gifDataSource
+                                     jpegDataSource:jpegDataSource
+                                      mp4DataSource:mp4DataSource
+                                        transaction:transaction];
+                    }];
+                remainder -= batchSize;
+                DDLogInfo(@"%@ sendFakeMediaMessages %zd / %zd", self.logTag, counter - remainder, counter);
+            }
+        });
+    }
+}
+
++ (void)sendFakeMediaMessages:(NSUInteger)counter
+                       thread:(TSThread *)thread
+                gifDataSource:(DataSource *)gifDataSource
+               jpegDataSource:(DataSource *)jpegDataSource
+                mp4DataSource:(DataSource *)mp4DataSource
+                  transaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    DDLogInfo(@"%@ sendFakeMediaMessages: %zd", self.logTag, counter);
+
+    for (NSUInteger i = 0; i < counter; i++) {
+
+        // Random time within last n years. Helpful for filling out a media gallery over time.
+        double yearsMillis = 4.0 * 365.0 * 24.0 * 60.0 * 60.0 * 1000.0;
```

Comment:
Instead, why not `yearsMillis = 4 * kYearInMs;`.   You'l need to define `kYearInMs` next to the existing constants, e.g. `kMonthInMs`.

Reply:
done!